pkgs/development/python-modules/pyopencl/default.nix

```diff
@@ -60,32 +64,29 @@ buildPythonPackage rec {
     pybind11
   ] ++ os-specific-buildInputs;
 
-  propagatedBuildInputs = [
-    appdirs
-    cffi
-    decorator
-    mako
+  dependencies = [
     numpy
     platformdirs
     pytools
-    six
   ];
 
   nativeCheckInputs = [ pytestCheckHook ];
 
   preBuild = ''
     export HOME=$(mktemp -d)
-    rm -rf pyopencl
```

Comment:
This feels related to https://github.com/NixOS/nixpkgs/issues/255262 . Try to `cd $out` in `preCheck` instead, and please link to the said issue in a comment.

Reply:
It also explains the error you receive